app_vue/src/utils/optimizeRouteHelper.js

```diff
@@ -1,39 +1,28 @@
 import axios from 'axios';
 
-const getGoogPayload = (sensorRouteList) => {
-  const waypointArr = [...sensorRouteList]; // ensures original variable isnt modified
-  const origin = waypointArr.shift(); // grab first sensor in route
-  const destination = waypointArr.pop(); // grab last sensor in route
-  
+const getGoogPayload = (selectedRouteList, originAddress, destinationAddress) => {
+  const origin = {
+    "address": originAddress
+  };
+  const destination = {
+    "address": destinationAddress
+  };
+  const waypointArr = [...selectedRouteList]; // ensures original variable isnt modified
   const intermediates = waypointArr.map(waypoint => {
     return {
-      "location":{
-        "latLng":{
+      "location": {
+        "latLng": {
           "latitude": waypoint.lat,
           "longitude": waypoint.long
         }
       }
     }
   });
 
-  const data = {
-    "origin": { // specify starting point
-      "location":{
-        "latLng":{
-          "latitude": origin.lat,
-          "longitude": origin.long
-        }
-      }
-    },
+  return {
+    origin,
     intermediates,
-    "destination":{ // specify ending point
-      "location":{
-        "latLng":{
-          "latitude": destination.lat,
-          "longitude": destination.long
-        }
-      }
-    },
+    destination,
```

Comment:
we are now taking in start and end points from user input, so updated file accordingly